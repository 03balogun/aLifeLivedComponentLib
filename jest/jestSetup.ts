/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
