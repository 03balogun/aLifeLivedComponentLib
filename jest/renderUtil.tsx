import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
