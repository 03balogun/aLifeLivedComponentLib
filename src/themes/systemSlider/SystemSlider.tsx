import {
  backgroundColor,
  BackgroundColorProps,
  border,
  BorderProps,
  BoxProps,
  createRestyleComponent,
  layout,
  LayoutProps,
  spacing,
  SpacingProps,
} from '@shopify/restyle';
import { Theme } from '../theme';
import Slider from '@react-native-community/slider';
import { SliderProps } from 'react-native';

export type StyledSliderProps = SliderProps &
  SpacingProps<Theme> &
  LayoutProps<Theme> &
  BorderProps<Theme> &
  BoxProps<Theme> &
  BackgroundColorProps<Theme>;

export const SystemSlider = createRestyleComponent<StyledSliderProps, Theme>(
  [spacing, layout, border, backgroundColor],
  Slider
);
