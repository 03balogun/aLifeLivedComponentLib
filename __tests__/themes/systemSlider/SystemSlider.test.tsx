import React from 'react';
import { render } from '../../../jest/renderUtil';
import { SystemSlider } from '../../../src/themes/systemSlider/SystemSlider';

const setup = () => render(<SystemSlider />);

describe('SystemSlider', () => {
  test('Should match snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });
});
