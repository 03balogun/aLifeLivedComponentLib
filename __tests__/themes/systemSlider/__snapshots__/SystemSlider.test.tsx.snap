// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemSlider Should match snapshot 1`] = `
<RNCSlider
  disabled={false}
  enabled={true}
  inverted={false}
  maximumValue={1}
  minimumValue={0}
  onChange={null}
  onRNCSliderAccessibilityAction={null}
  onRNCSliderSlidingComplete={null}
  onRNCSliderSlidingStart={null}
  onRNCSliderValueChange={null}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  step={0}
  style={
    Array [
      Array [
        Object {},
      ],
      Object {
        "height": 40,
      },
    ]
  }
  tapToSeek={false}
  value={0}
/>
`;
